src/backend/metal/src/factory.rs

```diff
@@ -247,6 +247,10 @@ impl core::Factory<Resources> for Factory {
                     .object_at(0)
                     .set_pixel_format(MTLPixelFormat::BGRA8Unorm_sRGB);
 
+                // We need fake depth attachments in case explicit writes to the depth buffer are required
+                pso_descriptor.set_depth_attachment_pixel_format(MTLPixelFormat::Depth24Unorm_Stencil8);
```

Comment:
I get an error for this:
```/Library/Caches/com.apple.xbs/Sources/Metal/Metal-85.82.1/Framework/MTLRenderPipeline.mm:1571: failed assertion `depthAttachmentPixelFormat is not a valid MTLPixelFormat.'```
Switching out for `MTLPixelFormat::Depth32_Stencil8` works though